klayout/drc/rule_decks/mim_b.drc

```diff
@@ -0,0 +1,149 @@
+  mimtm8a_l1.output('MIMTM.8a', 'MIMTM.8a : Minimum MIM cap area (defined by FuseTop area): 25µm²')
+  mimtm8a_l1.forget
+
+  # Rule MIMTM.8b: Maximum single MIM Cap area (Use multiple MIM caps in
+  ## parallel connection if bigger capacitors are required) (um2) is 10000µm.
+  logger.info('Executing rule MIMTM.8b')
+  mimtm8b_l1 = fusetop.with_area(10_000.um, nil).not_in(fusetop.with_area(10_000.um))
+  mimtm8b_l1.output('MIMTM.8b', "MIMTM.8b : Maximum single MIM Cap area (Use multiple MIM caps in
+                     parallel connection if bigger capacitors are required) (um2): 10000µm")
+  mimtm8b_l1.forget
+
+  # Rule MIMTM.9: Min. Via (Vian-1) spacing for sea of Via on MIM top plate is 0.5µm.
+  logger.info('Executing rule MIMTM.9')
+  mimtm9_l1 = top_via.inside(fusetop).space(0.5.um, euclidian)
+  mimtm9_l1.output('MIMTM.9', 'MIMTM.9 : Min. Via (Vian-1) spacing for sea of Via on MIM top plate: 0.5µm')
+  mimtm9_l1.forget
+
+  # Rule MIMTM.10: (a) There cannot be any Vian-2 touching MIM bottom plate Metaln-1.
+  ## (b) MIM bottom plate Metaln-1 can only be connected through the higher Via (Vian-1).
+  logger.info('Executing rule MIMTM.10')
+  mimtm10_l1 = topmin1_via.interacting(topmin1_metal_fusetop)
+  mimtm10_l1.output('MIMTM.10', "MIMTM.10 : (a) There cannot be any Vian-2 touching MIM bottom plate Metaln-1.
+                    (b) MIM bottom plate Metaln-1 can only be connected through the higher Via (Vian-1).")
+  mimtm10_l1.forget
+
+  # Rule MIMTM.11: Bottom plate of multiple MIM caps can be shared (for common nodes)
+  ## as long as total MIM area with that single common plate does not exceed MIMTM.8b rule.
+  logger.info('Executing rule MIMTM.11')
+  mimtm11_large_topmin1_metal = topmin1_metal_fusetop.with_area(10_000, nil)
+  mimtm11_large_topmin1_metal_violation = polygon_layer
+  mimtm11_large_topmin1_metal.data.each do |p|
+    mimtm11_topmin1_metal_polygon_layer = polygon_layer
+    mimtm11_topmin1_metal_polygon_layer.data.insert(p)
+    fuse_in_polygon = fusetop.and(mimtm11_topmin1_metal_polygon_layer)
+    next unless fuse_in_polygon.area > 10_000
+
+    mimtm11_bad_topmin1_metal_polygon = mimtm11_topmin1_metal_polygon_layer.interacting(fuse_in_polygon)
+    mimtm11_bad_topmin1_metal_polygon.data.each do |b|
+      b.num_points.positive? && mimtm11_large_topmin1_metal_violation.data.insert(b)
+    end
+  end
+  mimtm11_l1 = mimtm11_large_topmin1_metal_violation
+  mimtm11_l1.output('MIMTM.11',
+                    'MIMTM.11 : Bottom plate of multiple MIM caps can be shared (for common nodes)
+                     as long as total MIM area with that single common plate does not exceed MIMTM.8b rule.')
+  mimtm11_l1.forget
+  mimtm11_large_topmin1_metal.forget
+  mimtm11_large_topmin1_metal_violation.forget
+  topmin1_metal_fusetop.forget
+
+  # Rule MIMTM.12 is not a DRC check
+  # Please refer to https://gf180mcu-pdk.readthedocs.io/en/latest/physical_verification/design_manual/drm_10_4_2.html
```

Comment:
should we log something about the rule being ignored for completness?

Reply:
@proppy I think this will be confusing to report that. As these rules are not a real DRC rules. So we don't care about it.

@atorkmabrains Should we add logs for that ?

Reply:
Make sense, resolving.